src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/items/HopperJournal.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.wrappers.items;
+
+import net.neoforged.neoforge.transfer.transaction.SnapshotJournal;
+
+public final class HopperJournal extends SnapshotJournal<Boolean> {
```

Comment:
This whole thing could just be in the hopper. When writing patches, what matters is the number of hunks (consecutive-ish changed lines), not the length of each hunk.